static/app/views/dashboards/contexts/widgetSyncContext.tsx

```diff
@@ -0,0 +1,58 @@
+import type {ReactNode} from 'react';
+import type {EChartsType} from 'echarts';
+import * as echarts from 'echarts';
+
+import {uniqueId} from 'sentry/utils/guid';
+import {createDefinedContext} from 'sentry/utils/performance/contexts/utils';
+
+type RegistrationFunction = (chart: EChartsType) => void;
+
+interface WidgetSyncContext {
+  register: RegistrationFunction;
+}
+
+const [_WidgetSyncProvider, _useWidgetSyncContext, WidgetSyncContext] =
+  createDefinedContext<WidgetSyncContext>({
+    name: 'WidgetSyncContext',
+    strict: false,
+  });
+
+interface WidgetSyncContextProviderProps {
+  children: ReactNode;
+  groupName?: string;
+}
+
+export function WidgetSyncContextProvider({
+  children,
+  groupName = uniqueId(),
+}: WidgetSyncContextProviderProps) {
+  const register: RegistrationFunction = chart => {
+    chart.group = groupName;
+    echarts?.connect(groupName);
+  };
```

Comment:
Should this get wrapped with `useCallback`?

Reply:
Good point! just asking in case it would ever be used like that, I think it's fine to keep it like this